crates/driver/src/domain/competition/order/mod.rs

```diff
@@ -8,7 +8,8 @@ use crate::infra::{blockchain, Ethereum};
 
 /// Address used in place of an actual buy token address in an order which buys
 /// ETH.
-const BUY_ETH_ADDRESS: eth::TokenAddress = eth::TokenAddress(eth::H160([0xee; 20]));
+const BUY_ETH_ADDRESS: eth::TokenAddress =
+    eth::TokenAddress(eth::ContractAddress(eth::H160([0xee; 20])));
```

Comment:
This address only represents ETH as the `buy_token` from the SC point of view. So it depends on what we want this address to mean from the domain point of view.

Maybe it makes sense to:
```rust
pub enum TokenAddress {
    Erc20(ContractAddress),
    Eth,
}
```

Reply:
But wasn't this specific address picked because the `0xeee.eee` is already commonly used for `ETH`? In that case I would be fine with dropping it. Also I believe we are already using it in other contexts.